### PR TITLE
feat(docs): MDX components for recipes

### PR DESCRIPTION
**Pull request by YakovlevCoded**
Preview: https://deploy-preview-3178--cubejs-docs-staging.netlify.app/security

### CODE REVIEW

File: docs/src/components/GitHubCodeBlock/index.tsx [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+async function fetchCodeFromGitHub(url: string, setFile: (text: string) => void, part: string | null) {
+  const response = await fetch(url);
+  if (response.ok) {
+    let text = await response.text();
+
+    if (!part) {
+      setFile(text);
+    } else {
+      // @todo splice text
```
Comment (rchkv) [inline docs/src/components/GitHubCodeBlock/index.tsx:51]: This will be implemented in another PR?

File: docs/src/components/GitHubCodeBlock/index.tsx [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    
+    setTitle(title);
+    fetchCodeFromGitHub(
+      `https://raw.githubusercontent.com/${user}/${repo}/${branch}/${filePath}`, setFile, part
+    );
+  }, []);
+
+  return (
+    <div>
+      <h3>{title}</h3>
```
Comment (rchkv) [inline docs/src/components/GitHubCodeBlock/index.tsx:19]: We are already using `<h3>` on the pages (e.g. [here](https://raw.githubusercontent.com/cube-js/cube.js/master/docs/content/Configuration/Connecting-to-the-Database.md)). So I think it will be better to use `<h4>` for this title. Also, the `<h4>` title attracts less attention :)

Comment (YakovlevCoded) [reply docs/src/components/GitHubCodeBlock/index.tsx:19]: Yeah, it's just a draft. It's not ready to get a review :)
